commit 93b570464cce0079c15832cbb8ea17debd541585
Author: Jens Axboe <axboe@kernel.dk>
Date:   Wed Jan 10 09:06:23 2018 -0700

    null_blk: add option for managing IO timeouts
    
    Use the fault injection framework to provide a way for null_blk
    to configure timeouts. This only works for queue_mode 1 and 2,
    since the bio mode doesn't have code for tracking timeouts.
    
    Let's say you want to have a 10% chance of timing out every
    100,000 requests, and for 5 total timeouts, you could do:
    
    modprobe null_blk timeout="100000,10,0,5"
    
    This is useful for adding blktests to test that IO timeouts
    are handled appropriately.
    
    Signed-off-by: Jens Axboe <axboe@kernel.dk>

diff --git a/drivers/block/Kconfig b/drivers/block/Kconfig
index 40579d0cb3d1..622d9a2c8dae 100644
--- a/drivers/block/Kconfig
+++ b/drivers/block/Kconfig
@@ -19,6 +19,7 @@ if BLK_DEV
 config BLK_DEV_NULL_BLK
 	tristate "Null test block driver"
 	select CONFIGFS_FS
+	select FAULT_INJECTION
 
 config BLK_DEV_FD
 	tristate "Normal floppy disk support"